### PR TITLE
P4poller throws exception when server_tz string is not successfully resolved to a timezone offset

### PR DESCRIPTION
**Pull request by sphet**
I hit the case where I set a timezone and thought it was working, but spent some time figuring out why it wasn't and what the correct string for my location was.

## Contributor Checklist:

* [] I have updated the unit tests
* [X ] I have created a file in the `master/buildbot/newsfragment` directory (and read the `README.txt` in that directory)
* [X ] I have updated the appropriate documentation

### CODE REVIEW

**Comment by rodrigc** [comment]
Please add a unit test so that these lines are covered by codecov

**Comment by sphet** [comment]
@rodrigc I'll make that change but it's ridiculous to bar on it - the source file is full of hundreds of the same formatting. No disrespect but I'm more used to 'follow the existing pattern' not 'switch one line at a time' coding style.

**Comment by sphet** [comment]
@rodrigc How do I do a test that asserts and exception is thrown?

**Comment by rodrigc** [comment]
Look for unit tests under `master/buildbot/test/` which use `self.assertRaises()` to get some ideas

**Comment by sphet** [comment]
@rodrigc thanks - i'll look into it.

File: master/buildbot/changes/p4poller.py [hunk 1]
```diff
@@ -156,6 +156,8 @@ def __init__(self, p4port=None, p4user=None, p4passwd=None,
         self.use_tickets = use_tickets
         self.ticket_login_interval = ticket_login_interval
         self.server_tz = dateutil.tz.gettz(server_tz) if server_tz else None
+        if server_tz != None and self.server_tz == None:
+            raise P4PollerError("Failed to get timezone from server_tz string '{}'".format( server_tz) )
```
Comment (seankelly) [inline master/buildbot/changes/p4poller.py]: Need to remove the space after the `(` and before the trailing `)`.

File: master/buildbot/changes/p4poller.py [hunk 2]
```diff
@@ -156,6 +156,8 @@ def __init__(self, p4port=None, p4user=None, p4passwd=None,
         self.use_tickets = use_tickets
         self.ticket_login_interval = ticket_login_interval
         self.server_tz = dateutil.tz.gettz(server_tz) if server_tz else None
+        if server_tz != None and self.server_tz == None:
```
Comment (rodrigc) [inline master/buildbot/changes/p4poller.py]: pylint is complaining about this line:

```
buildbot/changes/p4poller.py:159 Comparison to None should be 'expr is None' [singleton-comparison]
```

**Pull request merged**